.github/workflows/build.yaml

```diff
@@ -3,97 +3,149 @@ on:
+      - uses: actions/checkout@v4
         with:
           submodules: recursive
-      - uses: ./.github/actions/setup/
-      - uses: ./.github/actions/setup-solana/
-      - uses: actions/cache@v2
-        name: Cache Cargo registry + index
-        id: cache-anchor
-        with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-          key: cargo-${{ runner.os }}-anchor-${{ hashFiles('**/Cargo.lock') }}
-      - run: cargo install --git https://github.com/coral-xyz/anchor --tag v0.28.0 anchor-cli --locked --force
-        working-directory: /
-      - name: Install latest nightly
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: nightly-2023-10-05-x86_64-unknown-linux-gnu
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
           components: rustfmt, clippy
+          toolchain: nightly-2023-10-05
+      - name: Install cargo-sort from crates.io
+        uses: baptiste0928/cargo-install@v3
+        with:
+          crate: cargo-sort
+      - run: cargo +nightly-2023-10-05 sort --workspace --check
+      - run: cargo +nightly-2023-10-05 fmt --all --check
+      - run: cargo +nightly-2023-10-05 clippy --all-features --all-targets --tests -- -D warnings
 
-      # --force is hack around cargo caching, figure out later
-      - name: Setup Rust tools
-        run: |
-          cargo +nightly-2023-10-05 install cargo-sort --force
-          cargo +nightly-2023-10-05 install cargo-udeps --force
-        shell: bash
-
-      - name: cargo sort
-        run: cargo +nightly-2023-10-05 sort --workspace --check
-        shell: bash
-
-      - name: cargo fmt
-        run: cargo +nightly-2023-10-05 fmt --all --check
-        shell: bash
+  udeps:
+    name: udeps
+    runs-on: big-runner-1
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          toolchain: nightly-2023-10-05
+      - name: Install cargo-udeps from crates.io
+        uses: baptiste0928/cargo-install@v3
+        with:
+          crate: cargo-udeps
+      - run: cargo +nightly-2023-10-05 udeps --all-features --all-targets --tests
 
-      - name: clippy
-        run: cargo +nightly-2023-10-05 clippy --all-features --all-targets --tests -- -D warnings
-        shell: bash
+  verified_build:
+    name: verified_build
+    runs-on: big-runner-1
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+      - name: Install solana-verify from crates.io
+        uses: baptiste0928/cargo-install@v3
+        with:
+          crate: solana-verify
+      - name: Install anchor-cli from crates.io
+        uses: baptiste0928/cargo-install@v3
+        with:
+          crate: anchor-cli
+          version: "0.29.0"
```

Comment:
this should be anchor 0.28

Reply:
CI not liking 0.28.0. unless you feel strongly about it, 0.29.0 seems fine as 0.28.0 never made it to crates.io